### PR TITLE
added section on assembly code and optimizations

### PR DESCRIPTION
**Pull request by campoy**

### CODE REVIEW

**Comment by campoy** [comment]
Hey @rakyll could you have a look?

File: 3-dynamic-analysis/3-profiling/1-benchmarks.md [hunk 1]
```diff
@@ -161,21 +161,121 @@ Sometimes your benchmarks might be way faster that you would expect. When
...
+	return int(float64(a) / float64(b))
+}
+func BenchmarkDiv(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		div(2, 3)
+	}
+}
+```
+
+If we run the benchmark we will get a surpsiring result:
```
Comment (bradleyfalzon) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: `surpsiring` should be `surprising`

File: 3-dynamic-analysis/3-profiling/1-benchmarks.md [hunk 2]
```diff
@@ -161,21 +161,121 @@ Sometimes your benchmarks might be way faster that you would expect. When
...
+BenchmarkDiv-8   	2000000000	         0.28 ns/op
+PASS
+ok  	github.com/campoy/go-tooling-workshop/3-dynamic-analysis/3-profiling	0.603s
+```
+
+This seems a bit too fast, even for a simple piece of code as the one we're benchmarking.
+We can see the generated code by adding `-gcflags "-S"`:
+
+```bash
+$ go test benchmarks_test.go -bench=Div -gcflags "-S" 2>& out.text
```
Comment (bradleyfalzon) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: Can I just clarify, what exactly were you trying to achieve with this redirect? `2>&` would redirect stderr to stdout, but then nothing is happening with stdout. If you intended to direct stdout and stderr to the file out.text you can do this in modern bash with just `&> out.txt` or more traditionally `> out.txt 2>&1`. 

I might be wrong, but running that command gives me:

```
go test benchmarks_test.go -bench=Div -gcflags "-S" 2>& out.text
-bash: out.text: ambiguous redirect
```

File: 3-dynamic-analysis/3-profiling/1-benchmarks.md [hunk 3]
```diff
@@ -161,21 +161,121 @@ Sometimes your benchmarks might be way faster that you would expect. When
...
+
+The problem here is that the function call to `div` has been inlined, and the inlined code
+has been deleted as it can be proven that its result is never used.
+
+You can see when a function call is inlined by using `gcflag "-m"`. Try it.
+
+A possible fix to this would be to use the `noinline` pragma, to avoid having `div` inlined.
+This would fix the problem, but would alter the results of our performance analysis too.
+
+A better idea is tostore the result of the function call
```
Comment (bradleyfalzon) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: `tostore` should be `to store`.

Comment (campoy) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: Thanks @bradleyfalzon, fixed

File: 3-dynamic-analysis/3-profiling/1-benchmarks.md (hunk 1 above)
Comment (campoy) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: done

File: 3-dynamic-analysis/3-profiling/1-benchmarks.md (hunk 2 above)
Comment (campoy) [inline 3-dynamic-analysis/3-profiling/1-benchmarks.md]: I need to figure out this, for now I'll go with your recommendation of `&>`
Thanks

**Pull request merged**